components/brave_federated_learning/brave_operational_profiling.cc

```diff
@@ -0,0 +1,192 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string>
+
+#include "brave/components/brave_federated_learning/brave_operational_profiling.h"
+
+#include "base/json/json_writer.h"
+#include "base/strings/string_util.h"
+#include "base/time/time.h"
+#include "base/timer/timer.h"
+#include "bat/ads/pref_names.h"
+#include "brave/components/brave_stats/browser/brave_stats_updater_util.h"
+#include "brave/components/brave_federated_learning/brave_operational_profiling_features.h"
+#include "brave/components/p3a/pref_names.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+#include <iostream>
+
+namespace brave {
+
+namespace {
+
+static constexpr char federatedLearningUrl[] = "https://fl.bravesoftware.com/";
+
+constexpr char kLastCheckedSlot[] = "brave.federated.last_checked_slot";
```

Comment:
I would make explicit that those are pref names, e.g. `kLastCheckedSlotPrefName`